modules/nextflow/src/main/groovy/nextflow/executor/FluxExecutor.groovy

```diff
@@ -0,0 +1,174 @@
+     */
+    protected List<String> getDirectives(TaskRun task, List<String> result) {
+        return result
+    }
+
+    // Flux does not require a special token or header
+    String getHeaderToken() { null }
+
+    /**
+     * The command line to submit this job
+     *
+     * @param task The {@link TaskRun} instance to submit for execution to the cluster
+     * @param scriptFile The file containing the job launcher script
+     * @return A list representing the submit command line
+     */
+    @Override
+    List<String> getSubmitCommandLine(TaskRun task, Path scriptFile ) {
+
+        List<String> result = ['flux', 'mini', 'submit']
+        result << '--setattr=cwd=' + quote(task.workDir)
+        result << '--job-name="' + getJobNameFor(task) + '"'
+        result << '--output=' + quote(task.workDir.resolve(TaskRun.CMD_LOG))  // -o OUTFILE
+
+        if( task.config.cpus > 1 ) {
+            result << '--cores=' + task.config.cpus.toString()
+        }
+
+        // Time limit in minutes when no units provided
+        if( task.config.time ) {
+            result << '--time-limit=' + task.config.getTime().format('mm')
+        }
+
+        // Flux does not support asking for specific memory
+        if( task.config.getMemory() ) {
+            log.debug "Custom memory request is not currently supported by Flux."
+        }
+
+        // the requested partition (a.k.a queue) name
+        if( task.config.queue ) {
+            result << '--queue=' + (task.config.queue.toString())
+        }
+
+        // Any extra cluster options the user wants!
+        // Options tokenized with ; akin to OarExecutor
+        if( task.config.clusterOptions ) {
+
+            // Split by space
+            for (String item : task.config.clusterOptions.toString().tokenize(' ')) {
+                if ( item ) {
+                    result << item.stripIndent().trim()
+                }
+            }
+        }
+        result << '/bin/bash' << scriptFile.getName()
+        return result
+    }
+
+    /**
+     * Parse the string returned by the {@code flux submit} command and extract the job ID string
+     *
+     * @param text The string returned when submitting the job
+     * @return The actual job ID string
+     */
+    @Override
+    def parseJobId(String text) {
+
+        // Parse the special "f" first
+        for( String line : text.readLines() ) {
+            def m = SUBMIT_REGEX.matcher(line)
+            if( m.find() ) {
+                return m.group(1).toString()
+            }
+        }
+
+        // Fall back to just a jobid
+        def id = text.trim()
+        if( id.isLong() )
+            return id
+
+        throw new IllegalStateException("Invalid Flux submit response:\n$text\n\n")
+    }
+
+    @Override
+    protected List<String> getKillCommand() { ['flux', 'job', 'cancel'] }
+
+    @Override
+    protected List<String> queueStatusCommand(Object queue) {
+
+        final result = ['flux', 'jobs', '--suppress-header']
```

Comment:
Suggestion: I worry a little here that if the default `flux jobs` output format changes for some reason, this section of the code could break without warning. Since `flux jobs` allows an output format to be specified, perhaps that could be used to ensure a static output format, e.g.

```java
        final result = [ 'flux', 'jobs', '-a',  '--suppress-header', '--format="{id.f58} {status_abbrev}"' ]
```

would just print a list of jobid/status pairs, e.g.:
```$ flux jobs -a --suppress-header --format="{id.f58} {status_abbrev}"
ƒaNehUumZ1 CD
ƒBgsJ5Yum1 CD
ƒBgkFZWouy CD
ƒBgMtbB71R CD
ƒ48VvoPC7h CD
ƒ28LvWXHXM CD
ƒ28Hp8Cfou CD
```

Note I also added `-a`, without which only "active" jobs are displayed (i.e. those that are pending or running)

Reply:
about `-a` nextflow uses this info also to determine when the task execution is complete. Ideally, it would be nice to include tasks completed in the last x minutes

Reply:
`flux jobs` does have a `--since` option:
```
      --since=WHEN           Include jobs that have become inactive since
                             WHEN. (implies -a if no other --filter option is
                             specified)
```

Reply:
Oh, and I should mention that `--since` can take an offset like `-30m` or even `15 minutes ago`

Reply:
awesome

Reply:
Okay I can change to since - what value goes into that? The function looks like it only takes "queue" as an argument?

Reply:
I think adding `--since -15m` should be enough